### PR TITLE
Build samples using Java 11 (too)

### LINKED ISSUE #72: Run build using Java 11
We currently don't run Lagom samples using AdoptOpenJDK 11. Need to add it there so that we can use the samples to validate Java 11 support.

### PR DESCRIPTION
**Pull request by ignasi35**
Fixes #72 

(this is WIP, I'm disabling Java 8 for the time being)

### CODE REVIEW

**Comment by ignasi35** [comment]
* `shopping-cart-java` (`sbt`) fails on Travis with:

```
java.lang.NoClassDefFoundError: sbt/ProjectReference
	at sbt.xMain.run(Main.scala:86)
	at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:109)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:128)
	at xsbt.boot.Launch$.run(Launch.scala:109)
	at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:35)
	at xsbt.boot.Launch$.launch(Launch.scala:117)
	at xsbt.boot.Launch$.apply(Launch.scala:18)
	at xsbt.boot.Boot$.runImpl(Boot.scala:56)
	at xsbt.boot.Boot$.main(Boot.scala:18)
	at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.lang.ClassNotFoundException: sbt.ProjectReference
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 10 more
Error during sbt execution: java.lang.NoClassDefFoundError: sbt/ProjectReference
The command "shopping-cart/travis-scripts/build.sh shopping-cart/shopping-cart-java sbt" exited with 1.
```

While on my local I get:

```
[info] Compiling 7 Java sources to /Users/ignasi/git/projects/lightbend/lagom/lagom-samples/shopping-cart/shopping-cart-java/shopping-cart/target/scala-2.12/classes ...
[warn] bootstrap class path not set in conjunction with -source 8
[error] warnings found and -Werror specified
[info] Done updating.
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[info] Compiling 2 Java sources to /Users/ignasi/git/projects/lightbend/lagom/lagom-samples/shopping-cart/shopping-cart-java/inventory/target/scala-2.12/classes ...
[warn] bootstrap class path not set in conjunction with -source 8
```

Which is also unexpected considering `shopping-cart-java` (`sbt`) doesn't set `-source 1.8` or `-target 1.8`:

```
javacOptions in (Compile,compile) ++= Seq("-Xlint:unchecked", "-Xlint:deprecation", "-parameters", "-Werror")
```

I've just checked and I don't know where that setting comes from but `sbt` decides it's is required. `sbt` also decides `-parameters` is not required:

```
sbt:shopping-cart-java> show Compile/javacOptions
[info] shopping-cart-api / Compile / javacOptions
[info] 	List()
[info] shopping-cart / Compile / javacOptions
[info] 	List(-g, -encoding, UTF-8, -source, 1.8, -target, 1.8, -parameters, -Xlint:unchecked, -Xlint:deprecation)
[info] inventory-api / Compile / javacOptions
[info] 	List()
[info] inventory / Compile / javacOptions
[info] 	List(-g, -encoding, UTF-8, -source, 1.8, -target, 1.8, -parameters, -Xlint:unchecked, -Xlint:deprecation)
[info] Compile / javacOptions
[info] 	List()
[success] Total time: 1 s, completed Jun 26, 2019, 8:41:30 PM
```

About `-parameters` being ignored, I think it's an issue affecting also `mixed-persistence-java-sbt` (which is why I needed to make some changes for Json serialisation to work).

**Comment by ignasi35** [comment]
/cc @eed3si9n

`sbt` projects in this repo don't seem to honor `javacOptions` when running on Java 11. 

This is probably related to the syntax `javacOptions in (Compile,compile)` but that's not an issue when running on Java 8. 🤷🏼‍♂️

**Comment by marcospereira** [comment]
Hey @ignasi35, should this target branch 1.6.x instead?

**Comment by ignasi35** [comment]
> Hey @ignasi35, should this target branch 1.6.x instead?

Indeed. Rebasing...

Thanks @marcospereira

**Comment by ignasi35** [comment]
> Hey @ignasi35, should this target branch 1.6.x instead?

On second thought... I'm not sure. We're already building Lagom itself `1.5.x` in Java 11.

**Comment by ignasi35** [comment]
> /cc @eed3si9n
> 
> `sbt` projects in this repo don't seem to honor `javacOptions` when running on Java 11.
> 
> This is probably related to the syntax `javacOptions in (Compile,compile)` but that's not an issue when running on Java 8. 🤷🏼‍♂️

After investigating a bit more, the issue is twofold:

* in most `sbt` projects here the syntax `javacOptions in xxx` is wrong. When fixing to `  javacOptions in Compile += "-parameters"` the issue goes away.
* `-api` projects don't have any `javacOptions` by default, while projects enabling the Lagom plugin includes [`defaultSettings`](https://github.com/lagom/lagom/blob/50ecfbf2e0d51fe24fdf6ad71157e1dff97106b9/dev/sbt-plugin/src/main/scala/com/lightbend/lagom/sbt/LagomSettings.scala#L13-L25). This is undocumented and generally these settings were ok but users of Java 11 will need different defaults (to be discussed in another issue).

**Comment by ignasi35** [comment]
The error: (in multiple samples)

```
java.lang.NoClassDefFoundError: sbt/ProjectReference
	at sbt.xMain.run(Main.scala:86)
	at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:109)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:128)
	at xsbt.boot.Launch$.run(Launch.scala:109)
	at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:35)
	at xsbt.boot.Launch$.launch(Launch.scala:117)
	at xsbt.boot.Launch$.apply(Launch.scala:18)
	at xsbt.boot.Boot$.runImpl(Boot.scala:56)
	at xsbt.boot.Boot$.main(Boot.scala:18)
	at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.lang.ClassNotFoundException: sbt.ProjectReference
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 10 more
Error during sbt execution: java.lang.NoClassDefFoundError: sbt/ProjectReference
The command "shopping-cart/travis-scripts/build.sh shopping-cart/shopping-cart-java sbt" exited with 1.
```

was caused by dirty caches in Travis.

**Comment by ignasi35** [comment]
marked `block-merge`.

I just realised I disabled builds for `Java 8` while working on Java 11. This is not ready for review yet.

File: .travis.yml [hunk 1]
```diff
@@ -3,77 +3,147 @@ language: scala
 before_install:
   # See https://github.com/travis-ci/travis-ci/issues/4629#issuecomment-239493916
   - rm ~/.m2/settings.xml
-  - curl -Ls https://git.io/jabba | bash && . ~/.jabba/jabba.sh
-
-install: jabba install adopt@1.8.192-12 && jabba use $_ && java -version
```
Comment (dwijnand) [inline .travis.yml:8]: Keep this, but make it

```yaml
install: jabba install "$TRAVIS_JDK" && jabba use $_ && java -Xmx32m -version
```

with:

* TRAVIS_JDK defaulted as `TRAVIS_JDK=adopt@1.8.202-08` in `env.matrix`, and
* `env: TRAVIS_JDK=adopt@1.11.0-2` in the `test-java11` jobs.

So like Play's setup: https://github.com/playframework/playframework/blob/master/.travis.yml

At this point in time I'm not a fan of the creative indirection of `bin/jabbaLib` functions... but that's just my gut reaction.  Perhaps I could come around (and maybe even promote it) 😄

Comment (ignasi35) [reply .travis.yml:8]: I tried to bring this `.travis.yml` close to the one in https://github.com/playframework/play-samples but this build uses `stages` for deployment jobs and my head exploded a bit.

**Comment by ignasi35** [comment]
I'll redo this PR pushing the branch on `upstream` so `deploy` stages in the build also run. See #75

**Pull request closed without merging**